src/test/scala/mesosphere/marathon/api/v2/validation/AppValidationTest.scala

```diff
@@ -277,4 +277,65 @@ class AppValidationTest extends UnitTest with ValidationTestLike {
     }
 
   }
+
+  "health check validation" when {
+    val allowedProtocols = Set(
```

Comment:
Shouldn't we make sure to test behavior for all diallowed protocols as well?
`AppHealthCheckProtocol` is a sealed trait, so this could be done with a little help from the compiler :)

Reply:
Yep, I am not sure either ... I can definitely create manual TableDriven test...

Reply:
Sorry – I was referring to the map but I wasn't aware it's private.
The compiler help would probably be little over engineered: a partial function that triggers the tests, and a verification based on the map to check that the PF is defined for all available values :)